### PR TITLE
Subscription: remove restrictions on topic when subscribing to TsFile

### PR DESCRIPTION
**Pull request by VGalaxies**
As stated in the title, this PR also includes the following minor changes:

- Fixed the data insertion logic in the receiver env of the consumer group IT.
- Added live/snapshot parameters to the pipe source mode, corresponding to subscribe/query.
- Maintenance of current subscription topics information by the consumer and some optimizations.

### CODE REVIEW

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/conf/CommonConfig.java [hunk 1]
```diff
@@ -251,13 +253,13 @@ public class CommonConfig {
 
   private int subscriptionSubtaskExecutorMaxThreadNum =
       Math.min(5, Math.max(1, Runtime.getRuntime().availableProcessors() / 2));
-  private int subscriptionMaxTabletsPerPrefetching = 64;
-  private int subscriptionMaxTabletsSizeInBytesPerPrefetching = 8388608; // 8MB
+  private int subscriptionPrefetchBatchMaxDelayInMs = 5000; // 5s
+  private long subscriptionPrefetchBatchMaxSizeInBytes = 80 * MB;
```
Comment (Caideyipi) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/conf/CommonConfig.java:257]: In pipes, we use 1s and 16MB by default in tablet plain batch. Is it OK to use 5s and 80MB (which are the default settings of tsFile batch) in both batches?

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/subscription/event/SubscriptionEvent.java [hunk 2]
```diff
@@ -19,45 +19,94 @@
...
+
   private static final long INVALID_TIMESTAMP = -1;
 
-  protected final List<EnrichedEvent> enrichedEvents;
-  protected final SubscriptionPollResponse response;
+  private final SubscriptionPipeEvents pipeEvents;
+  private final SubscriptionPollResponse[] responses;
+  private int currentResponseIndex;
+
+  private final ByteBuffer[] byteBuffers; // serialized responses
```
Comment (Caideyipi) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/subscription/event/SubscriptionEvent.java:58]: Is this byteBuffer even used?...

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/subscription/event/SubscriptionEvent.java [hunk 3]
```diff
@@ -116,39 +178,169 @@ public String getLastPolledConsumerId() {
...
+      byteBuffers[currentResponseIndex] = null;
     }
-    return SubscriptionPollResponse.serialize(response);
   }
 
-  public ByteBuffer getByteBuffer() {
-    return byteBuffer;
+  /////////////////////////////// tsfile ///////////////////////////////
+
+  public static @NonNull SubscriptionEvent generateSubscriptionEventWithInitPayload(
```
Comment (Caideyipi) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/subscription/event/SubscriptionEvent.java:196]: Is it necessary to put a method here specially?

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/subscription/event/SubscriptionEvent.java [hunk 4]
```diff
@@ -116,39 +178,169 @@ public String getLastPolledConsumerId() {
 
   //////////////////////////// byte buffer ////////////////////////////
 
-  public ByteBuffer serialize() throws IOException {
-    if (Objects.nonNull(byteBuffer)) {
-      return byteBuffer;
+  public void resetByteBuffer(final boolean resetAll) {
```
Comment (Caideyipi) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/subscription/event/SubscriptionEvent.java:181]: It seems that there is no need to call "BinaryCache" to invoke this method..

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/subscription/event/SubscriptionEvent.java [hunk 5]
```diff
@@ -116,39 +178,169 @@ public String getLastPolledConsumerId() {
...
+    if (resetAll) {
+      Arrays.stream(responses)
+          .forEach((response -> SubscriptionEventBinaryCache.getInstance().invalidate(response)));
+      // maybe friendly for gc
+      Arrays.fill(byteBuffers, null);
```
Comment (Caideyipi) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/subscription/event/SubscriptionEvent.java:186]: Be sure to check the byte buffer...

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/subscription/event/SubscriptionEventBinaryCache.java [hunk 6]
```diff
@@ -37,19 +38,21 @@ public class SubscriptionEventBinaryCache {
...
   private static final Logger LOGGER = LoggerFactory.getLogger(SubscriptionEventBinaryCache.class);
 
-  private final PipeMemoryBlock allocatedMemoryBlock;
-
   private final AtomicDouble memoryUsageCheatFactor = new AtomicDouble(1);
 
-  private final LoadingCache<SubscriptionEvent, ByteBuffer> cache;
+  private final LoadingCache<SubscriptionPollResponse, ByteBuffer> cache;
 
   public ByteBuffer serialize(final SubscriptionEvent event) throws IOException {
```
Comment (Caideyipi) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/subscription/event/SubscriptionEventBinaryCache.java:45]: Is it better to delete the interface and use serialize(response) only?

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/subscription/event/SubscriptionEventBinaryCache.java [hunk 7]
```diff
@@ -59,21 +62,31 @@ public ByteBuffer serialize(final SubscriptionEvent event) throws IOException {
    * @return true -> byte buffer is not null
    */
   public boolean trySerialize(final SubscriptionEvent event) {
```
Comment (Caideyipi) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/subscription/event/SubscriptionEventBinaryCache.java:64]: The same as above

File: iotdb-client/session/src/main/java/org/apache/iotdb/session/subscription/consumer/SubscriptionPushConsumer.java [hunk 8]
```diff
@@ -310,4 +312,27 @@ public SubscriptionPushConsumer buildPushConsumer() {
...
+  @Override
+  protected Map<String, String> coreReportMessage() {
+    final Map<String, String> coreReportMessage = super.coreReportMessage();
+    coreReportMessage.put("ackStrategy", ackStrategy.toString());
+    return coreReportMessage;
+  }
+
+  @Override
+  protected Map<String, String> allReportMessage() {
+    final Map<String, String> coreReportMessage = super.allReportMessage();
```
[the remark this answers is not in the document]
Comment (VGalaxies) [reply iotdb-client/session/src/main/java/org/apache/iotdb/session/subscription/consumer/SubscriptionPushConsumer.java:332]: referenced `org.apache.iotdb.commons.pipe.event.EnrichedEvent#coreReportMessage`, perhaps a better name could be found...

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/conf/CommonConfig.java (hunk 1 above)
Comment (VGalaxies) [reply iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/conf/CommonConfig.java:257]: Previously overlooked this issue. Now it stays consistent with pipe.

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/subscription/event/SubscriptionEvent.java (hunk 2 above)
Comment (VGalaxies) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/subscription/event/SubscriptionEvent.java:58]: The logic here needs to be improved...

**Review by SteveYurongSu — APPROVED** [review]

**Pull request merged**